### ISSUE TITLE
Issue a compiler warning when using the is operator to compare a variable of a non-nullable reference type with its declared type.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#34431]
<p>I am currently debugging a library following some refactoring where a similar expression was used by mistake. I feel it would be helpful if this would issue a compiler warning to make the problem easier to find, similar to how the same thing with a non-nullable value type issues a warning.</p><p>The following is example C# code that would issue this warning:</p><pre>#nullable enable
bool isString(string s) =&gt; s is string;</pre><p>Of course, the is operator would return false if s were null, but s is declared non-nullable.</p>

_This issue has been moved from https://developercommunity.visualstudio.com/content/idea/502487/issue-a-compiler-warning-when-using-the-is-operato.html
VSTS ticketId: 825470_
_These are the original issue comments:_

Jane Wu [MSFT] on 3/22/2019, 02:02 AM (3 days ago): <p>Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further.&nbsp;We'll provide an update once the issue has been triaged by the product team.</p>

**Comment by gafter** [comment]
We already decided not to give diagnostics when the result of null checks are believed known.  This code is providing the service of checking whether its argument is null or not, and would be helpful asserting when the annotation on a variable in the caller is incorrect, e.g.
``` c#
void M(string s)
{
    Debug.Assert(isString(arg));
```
